include/public/marco/Codegen/Lowering/Distance/EditDistance.h

```diff
@@ -0,0 +1,57 @@
+#ifndef MARCO_CODEGEN_LOWERING_DISTANCE_EDITDISTANCE_H
+#define MARCO_CODEGEN_LOWERING_DISTANCE_EDITDISTANCE_H
+
+#include <array>
+#include <cassert>
+#include <optional>
+#include <string>
+
+namespace marco::codegen::lowering
+{
+  class EditDistance
```

Comment:
It is a collection of functions, and parameters that are relative to the concept of "edit distance", so they were all put together into a single class. Should I remove the class and create an EditDistance namespace instead?

Reply:
Classes are made to be instantiated, otherwise they serve no purpose. Remove the class and let the methods live directly inside the namespace (`marco::codegen::lowering` is enough).